packages/web3-providers/src/opensea/index.ts

```diff
@@ -1 +1,320 @@
+
+import BigNumber from 'bignumber.js'
+import {
+    ChainId,
+    createERC721ContractDetailed,
+    createERC721Token,
+    ERC721TokenDetailed,
+    EthereumTokenType,
+    isSameAddress,
+} from '@masknet/web3-shared-evm'
+async function fetchAsset(url: string, chainId: ChainId) {
+    if (![ChainId.Mainnet, ChainId.Rinkeby].includes(chainId)) return
+
+    const response = await (
+        await fetch(url, {
+            method: 'GET',
+            mode: 'cors',
+            headers: {
+                'x-api-key': OPENSEA_API_KEY,
+            },
+        })
+    ).json()
+
+    return response
+}
+
+export async function getNFTsPaged(from: string, opts: { chainId?: ChainId; page?: number; size?: number }) {
+    const { chainId = ChainId.Mainnet, page = 0, size = 50 } = opts
+    const params = new URLSearchParams()
+    params.append('owner', from.toLowerCase())
+    params.append('limit', String(size))
+    params.append('offset', String(size * page))
+
+    const asset = await fetchAsset(`${OpenSea_API_URL}/api/v1/assets?${params.toString()}`, chainId)
+    if (!asset) return []
+
+    return asset.assets
+        .filter(
+            (x: OpenSeaResponse) =>
+                ['non-fungible', 'semi-fungible'].includes(x.asset_contract.type) ||
+                ['ERC721', 'ERC1155'].includes(x.asset_contract.schema_name),
+        )
+        .map((asset: OpenSeaResponse) => createERC721TokenAsset(from, asset.token_id, chainId, asset))
+}
+
+function createERC721ContractDetailedFromAssetContract(
+    address: string,
+    chainId: ChainId,
+    assetContract?: OpenSeaAssetContract,
+) {
+    return createERC721ContractDetailed(chainId, address, assetContract?.name ?? 'unknown name', '')
+}
+
+function createERC721TokenAsset(address: string, tokenId: string, chainId: ChainId, asset?: OpenSeaResponse) {
+    return createERC721Token(
+        createERC721ContractDetailed(chainId, address, asset?.name ?? 'unknown name', ''),
+        {
+            name: asset?.name ?? asset?.asset_contract.name ?? 'unknown name',
+            description: asset?.description ?? '',
+            mediaUrl: asset?.image_url_original ?? asset?.image_url ?? asset?.image_preview_url ?? '',
+            owner: asset?.owner.address ?? '',
+        },
+        tokenId,
+    )
+}
+
+async function _getAsset(address: string, tokenId: string, chainId: ChainId) {
+    const asset = (await fetchAsset(
+        urlcat(`${OpenSea_API_URL}/api/v1/asset/:address/:tokenId`, { address, tokenId }),
+        chainId,
+    )) as OpenSeaResponse
+    return asset
+}
+
+export async function getContract(address: string, chainId = ChainId.Mainnet) {
+    const assetContract = await fetchAsset(`${OpenSea_API_URL}/api/v1/asset_contract/${address}`, chainId)
+
+    return createERC721ContractDetailedFromAssetContract(address, chainId, assetContract)
+}
+
+export async function getNFT(address: string, tokenId: string, chainId: ChainId) {
```

Comment:
Why do some functions with an optional `chainId` parameter, some are not? Please keep them the same design.